apps/client/src/app/pages/portfolio/allocations/allocations-page.component.ts

```diff
@@ -490,7 +490,32 @@ export class AllocationsPageComponent implements OnDestroy, OnInit {
           name,
           allocationInPercentage:
             this.totalValueInEtf > 0 ? value / this.totalValueInEtf : 0,
-          valueInBaseCurrency: value
+          valueInBaseCurrency: value,
+          parents: Object.entries(this.portfolioDetails.holdings)
+            .map(([symbol, holding]) => {
+              if (holding.holdings) {
+                const parentHoldings = holding.holdings;
+                for (const index in parentHoldings) {
```

Comment:
Would you also prefer to run the parents loop before the `hasImpersonationId` check so support this addition in safe mode as well? (without actual currency values)

Reply:
I can update the code to use `find`, however, the current topHoldingsMap loop does not include only includes the name and value, nothing more, so I cannot compare symbol or dataSource unless I add it to the map.

Reply:
In the latest commit I've converted it to use `find`. However, the comparison is still done by name since the other info is not available. in the topHoldingsMap.